src/connector/src/parser/avro/util.rs

```diff
@@ -152,10 +164,152 @@ fn avro_type_mapping(schema: &Schema) -> ConnectorResult<DataType> {
                 bail!("unsupported Avro type: {:?}", schema);
             }
         }
-        Schema::Map(_) | Schema::Null | Schema::Fixed(_) | Schema::Uuid => {
+        Schema::Map(value_schema) => {
+            // TODO: support native map type
+            match map_handling {
+                Some(MapHandling::Jsonb) => {
+                    if supported_avro_to_json_type(value_schema) {
+                        DataType::Jsonb
+                    } else {
+                        bail!(
+                            "unsupported Avro type, cannot convert map to jsonb: {:?}",
+                            schema
+                        )
+                    }
+                }
+                None => {
+                    bail!("`map.handling.mode` not specified in ENCODE AVRO (...). Currently supported modes: `jsonb`")
+                }
+            }
+        }
+        Schema::Null | Schema::Fixed(_) | Schema::Uuid => {
             bail!("unsupported Avro type: {:?}", schema)
         }
     };
 
     Ok(data_type)
 }
+
+/// Check for [`avro_to_jsonb`]
+fn supported_avro_to_json_type(schema: &Schema) -> bool {
+    match schema {
+        Schema::Null | Schema::Boolean | Schema::Int | Schema::String => true,
+
+        Schema::Map(value_schema) | Schema::Array(value_schema) => {
+            supported_avro_to_json_type(value_schema)
+        }
+        Schema::Union(union_schema) => union_schema
+            .variants()
+            .iter()
+            .all(supported_avro_to_json_type),
```

Comment:
There is a caveat when converting avro union to jsonb: the former may be tagged (like rust enum) but the latter may be untagged (like c union). When the union consists of multiple `record`s, it is possible to distinguish which variant is active in avro, but in json they will all become jsonb objects and indistinguishable.

Can we disallow union -> jsonb as well?